src/schema.ts

```diff
@@ -131,44 +164,44 @@ export function build(data: any, opt: QueryConfig = {}): Schema {
   return new Schema(fieldSchemas);
 }
 
-export class Schema {
+export class Schema implements TableSchema {
   private _fieldSchemas: FieldSchema[];
   private _fieldSchemaIndex: {[field: string]: FieldSchema};
 
   constructor(fieldSchemas: FieldSchema[]) {
     this._fieldSchemas = fieldSchemas;
     this._fieldSchemaIndex = fieldSchemas.reduce((m, fieldSchema: FieldSchema) => {
-      m[fieldSchema.field] = fieldSchema;
+      m[fieldSchema.name] = fieldSchema;
       return m;
     }, {});
   }
 
   /** @return a list of the field names (for enumerating). */
-  public fields() {
-    return this._fieldSchemas.map((fieldSchema) => fieldSchema.field);
+  public names() {
+    return this._fieldSchemas.map((fieldSchema) => fieldSchema.name);
   }
 
   /** @return a list of FieldSchemas */
-  public get fieldSchemas() {
+  public get fields() {
```

Comment:
Now I wonder why we can override `fields` array as a getter.  This seems to cloak the `fields` that comes with TableSchema.  

Given this and [the comment here](https://github.com/vega/compassql/pull/379/files#r121759269), I start wondering if we better make Schema encapsulate TableSchema rather than `implements` it as discussed [here](https://github.com/vega/compassql/pull/379/files#r121749686)?